src/System.Memory/src/System.Memory.csproj

```
@@ -0,0 +1,77 @@
+    <Compile Include="$(CommonPath)\CoreLib\System\Buffers\MemoryHandle.cs" Link="Common\CoreLib\System\Buffers\MemoryHandle.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\Buffers\MemoryManager.cs" Link="Common\CoreLib\System\Buffers\MemoryManager.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\Memory.cs" Link="Common\CoreLib\System\Memory.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\MemoryDebugView.cs" Link="Common\CoreLib\System\MemoryDebugView.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\MemoryExtensions.cs" Link="Common\CoreLib\System\MemoryExtensions.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\ReadOnlyMemory.cs" Link="Common\CoreLib\System\ReadOnlyMemory.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\ReadOnlySpan.cs" Link="Common\CoreLib\System\ReadOnlySpan.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\Runtime\InteropServices\MemoryMarshal.cs" Link="Common\CoreLib\System\Runtime\InteropServices\MemoryMarshal.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\Span.cs" Link="Common\CoreLib\System\Span.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\SpanDebugView.cs" Link="Common\CoreLib\System\SpanDebugView.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\SpanHelpers.BinarySearch.cs" Link="Common\CoreLib\System\SpanHelpers.BinarySearch.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\SpanHelpers.Byte.cs" Link="Common\CoreLib\System\SpanHelpers.Byte.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\SpanHelpers.Char.cs" Link="Common\CoreLib\System\SpanHelpers.Char.cs" />
+    <Compile Include="$(CommonPath)\CoreLib\System\SpanHelpers.T.cs" Link="Common\CoreLib\System\SpanHelpers.T.cs" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true' and $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) != '.NETFramework'">
+    <Compile Include="System\Buffers\ReadOnlySequenceSegment.cs" />
+    <Compile Include="System\Buffers\StandardFormat.cs" />
+    <Compile Include="System\MemoryExtensions.Portable.cs" />
+    <Compile Include="System\NUint.cs" />
+    <Compile Include="System\Pinnable.cs" />
+    <Compile Include="System\ReadOnlySpan.Portable.cs" />
+    <Compile Include="System\Runtime\InteropServices\MemoryMarshal.Portable.cs" />
+    <Compile Include="System\Span.Portable.cs" />
+    <Compile Include="System\SpanHelpers.cs" />
+    <Compile Include="System\ThrowHelper.cs" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\System.Buffers\src\System.Buffers.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(IsPartialFacadeAssembly)' != 'true'">
+    <PackageReference Include="System.Numerics.Vectors" />
+    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" />
+  </ItemGroup>
+
+
+  <ItemGroup Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETFramework'">
```

The function call isn't necessary for item conditions. Just use the TFI property instead:

```suggestion
  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
```

---

Why are they not necessary for item conditions but yes for everything else?

---

They are only necessary for properties / property group conditions as TFI/TFV aren't available in the project body because the Microsoft.NET.Sdk.targets sets them. They are available for properties in a Directory.Build.targets file, just not before that. Items are evaluated after properties so TFI/TFV is always available.